minseo/service/userService.js

```diff
@@ -1,29 +1,47 @@
 const userDao = require('../models/userDao');
+const bcrypt = require('bcrypt');
+const saltRounds = 8;
+const jwt = require('jsonwebtoken');
 
-const signUp = async (name, email, password, profileImage) => {
+const signUp = async (name, email, password) => {
   const pwValidation = new RegExp(
     '^(?=.*[A-Za-z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,20})'
   );
 
   const emailValidation = new RegExp(
-    '/^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i'
+    "([!#-'*+/-9=?A-Z^-~-]+(.[!#-'*+/-9=?A-Z^-~-]+)*|\"([]!#-[^-~ \t]|(\\[\t -~]))+\")@([!#-'*+/-9=?A-Z^-~-]+(.[!#-'*+/-9=?A-Z^-~-]+)*|[[\t -Z^-~]*])"
   );
 
   if (!pwValidation.test(password) || !emailValidation.test(email)) {
     const err = new Error('PASSWORD_OR_EMAIL_IS_NOT_VALID');
-    err.statusCode = 400;
+    err.statusCode = 409;
     throw err;
   }
 
-  const createUser = await userDao.createUser(
-    name,
-    email,
-    password,
-    profileImage
-  );
+  const hashPassword = bcrypt.hashSync(password, saltRounds);
+
+  const createUser = await userDao.createUser(name, email, hashPassword);
   return createUser;
 };
 
+const signIn = async (email, password) => {
+  const user = await userDao.getUserByEmail(email);
+  const match = await bcrypt.compare(password, user.password);
+
+  if (!match) {
+    const err = new Error('NOT_USER');
+    err.statusCode = 409;
+    throw err;
+  }
+
+  const id = user.id;
+  const payLoad = { email: email, id: id };
```

Comment:
email도 개인 정보가 될 수 있으므로 특별한 이유가 아니면 토큰에 포함 안 시키는 것이 좋습니다.
```suggestion
  const payLoad = {id: id };
```